streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -172,21 +173,28 @@ public ProcessorNode build() {
         private final Pattern pattern;
         private final Deserializer<?> keyDeserializer;
         private final Deserializer<?> valDeserializer;
-
-        private SourceNodeFactory(String name, String[] topics, Pattern pattern, Deserializer<?> keyDeserializer, Deserializer<?> valDeserializer) {
+        private final TimestampExtractor timestampExtractor;
+
+        private SourceNodeFactory(final String name,
+                                  final String[] topics,
+                                  final Pattern pattern,
+                                  final TimestampExtractor timestampExtractor,
+                                  final Deserializer<?> keyDeserializer,
+                                  final Deserializer<?> valDeserializer) {
             super(name);
             this.topics = topics != null ? Arrays.asList(topics) : null;
             this.pattern = pattern;
             this.keyDeserializer = keyDeserializer;
             this.valDeserializer = valDeserializer;
+            this.timestampExtractor = timestampExtractor;
         }
 
         List<String> getTopics(Collection<String> subscribedTopics) {
             // if it is subscribed via patterns, it is possible that the topic metadata has not been updated
             // yet and hence the map from source node to topics is stale, in this case we put the pattern as a place holder;
             // this should only happen for debugging since during runtime this function should always be called after the metadata has updated.
             if (subscribedTopics.isEmpty())
-                return Collections.singletonList("Pattern[" + pattern + "]");
+                return Collections.singletonList("" + pattern + "");
```

Comment:
@guozhangwang actually `"Pattern[ ]"` string is just cosmetic part of the code snippet 
`"Pattern[" + pattern + "]"`. We don't use `"Pattern"` string (anywhere in the code) for detecting if it is a single subscribed topic or subscribed pattern.